src/Features/Core/Portable/DesignerAttribute/DesignerAttributeComputer.cs

```diff
@@ -4,73 +4,81 @@
 
 using System;
 using System.Collections.Concurrent;
+using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.ErrorReporting;
 using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.CodeAnalysis.Shared.Extensions;
-using Microsoft.CodeAnalysis.SolutionCrawler;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.DesignerAttribute
 {
-    internal abstract class AbstractDesignerAttributeIncrementalAnalyzer : IncrementalAnalyzerBase
+    internal sealed partial class DesignerAttributeComputer
```

Comment:
there are two core parts to the DesignAttribute scanning.  This 'computer' object which is effectively just the feature-level logic for scanning a solution finding the types with that attribute, and caching the result.  THis object is held alive on the OOP side and is periodically queried by the VS side when the solution changes.  The VS side then gets teh simple data notifications and then is responsible for plugging that into the VS apis to present the data.

Reply:
the major change of this type is moving away from an incremental anlayzer and jsut being a simple type that does its own walking of hte solution, notifying a passed in 'callback' of what it has found.  the VS side is responsible for determining when to call into this.